common/src/Model/Brush.cpp

```diff
@@ -947,8 +970,23 @@ namespace TrenchBroom {
             assert(remaining.vertexCount() + moving.vertexCount() == vertexCount());
             
             // Will vertices be removed?
-            if (!allowVertexRemoval && result.vertexCount() < m_geometry->vertexCount())
-                return false;
+            if (!allowVertexRemoval) {
+                if (result.vertexCount() < m_geometry->vertexCount())
```

Comment:
Dammit, I've been thinking about vertex editing for years now, and there's always something I forget. Yup, you're right.